src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
If it's not documented that it's not a change, so also no Breaking Change :>
Seriously though, I thought this change would only align ng.$animate with ngAnimate.$animate, but it turns out cancelling an actual animation resolves the promise. 
The reason why I change this code is because otherwise .done isn't called with `false`, so some switch tests fail. The problem is that these ngSwitch tests are "synthetic" in the sense that they never complete the "animations" they start because the rAFs aren't flushed. I think if we did that, then we didn't have to do this change. However, we would have to insert raF flushes in about 8 ngSwitch tests where they actually don't need to be there from considering the test's scope (these are not animation tests).